model-server/src/main/kotlin/org/modelix/model/server/handlers/ContentExplorer.kt

```diff
@@ -138,8 +136,12 @@ class ContentExplorer(private val client: IModelClient, private val repoManager:
             div("nameField") {
                 attributes["data-nodeid"] = node.nodeId.toString()
                 b {
+                    val namePropertyUID = BuiltinLanguages.jetbrains_mps_lang_core.INamedConcept.name.getUID()
+                    val namedConceptRole = node.getPropertyRoles().find { it.endsWith(namePropertyUID) }
                     if (node.getPropertyRoles().contains("name")) {
                         +"${node.getPropertyValue("name")}"
+                    } else if (namedConceptRole != null) {
+                        +"${node.getPropertyValue(namedConceptRole)}"
```

Comment:
Is this ordering sensible? If something is a named concept, shouldn't we take that value with preference?

Reply:
Maybe adding a comment on why ordering was chosen this way would have been good, too. The next one might be wondering again.